### ISSUE TITLE
App crashing when being put into background with MapView component mounted

### ISSUE THREAD

**Issue by shorsher** [react-native-maps/react-native-maps#3650]
# Bug report

## Summary
Hi! I have recently ejected from expo, and ran into a new issue with maps. If I'm on a screen with a <MapView> component, whenever I place my app into the background it crashes. Running the application on iOS via xcode. The error in Xcode is shown as: `Thread 26: EXC_BAD_ACCESS (code=1, address=0x10)`. I have tried to uninstall and reinstall react-native-maps since I have ejected from expo, but still having the same issue. I am able to reproduce with a fresh react-native (non-expo), application. 

## Environment info

`react-native info` output:

```bash
System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 933.69 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.15.0 - /var/folders/dj/fvcdsjcn3_s0g6cxb5kcq2wr0000gn/T/fnm_multishell_13416_1607353176693/bin/node
    Yarn: Not Found
    npm: 6.14.8 - /var/folders/dj/fvcdsjcn3_s0g6cxb5kcq2wr0000gn/T/fnm_multishell_13416_1607353176693/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.10.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
    Android SDK:
      API Levels: 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.2
      System Images: android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6858069
    Xcode: 12.2/12B45b - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_275 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: ~0.63.3 => 0.63.4
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
``` bash
    "react-native-maps": "0.27.1",
```

## Steps to reproduce

1. Start a fresh react native application, install react-native-maps, use example map code from README.
2. Run application
3. Place application in background. 

Describe what you expected to happen:

1. When application is reopened, everything is still working. (app not frozen, crashed, etc.)

Describe what you actually happens:

1. When put into background, app crashes with `Thread 51: EXC_BAD_ACCESS (code=1, address=0x10)`

## Reproducible sample code

gist: https://gist.github.com/shorsher/5049b8ae3a27e113d6620a4935ff7362

**Comment by shorsher** [comment]
After some further debugging, it seems hardcoding a height and width prevents the crashing from happening. But in my actual application I am using:

```javascript
        width: Dimensions.get('window').width,
        height: Dimensions.get('window').height,
```
and the crashing is still occurring.

If I use something like 400/400, everything seems to be fine.

**Comment by fivecar** [comment]
@shorsher: I believe this is still an issue, even in 0.29.3. I've filed #3957. Note, for your app, the results I get when testing: that setting an explicit width/height merely reduces the reliability of the repro, but the crash is still there. If you absolutely need to set a consistent width/height, and want a more consistent repro, here's what I've found tends to work (for me):

1. First set it to `flex:1`. Build & run the app, and switch out so that it crashes.
2. Now set it to `width: 400, height:400`, or whatever you like, and once again build & run.

Following the steps above, I'm able to get a nearly consistent repro even setting an explicit width/height. Wanted to let you know in case it might still be a problem in your app.
